### ISSUE TITLE
All contacts show *Last modified seconds ago*

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/contacts#3414]
### Describe the bug

Last modified date, if known, should work reliably

### Steps to reproduce

1. Open the contacts app
2. Open any contact (can be your own, recently contacted or system address book)
3. Scroll to the bottom

### Expected behavior

Real last modified date

### Actual behavior

![image](https://github.com/nextcloud/contacts/assets/1374172/df6e3078-1e10-4a67-b321-af7ac5bd2ba6)

### Contact version

5.3

### PHP engine version

None

### Web server

None

### Database

None

**Comment by ChristophWurst** [comment]
I still see it with 27 RC2. It happens with my own contacts, recently contacted and system contacts

**Comment by ChristophWurst** [comment]
I tested again and this only happens with system contacts. A contact in a real address book shows *Last edited 5 months ago*

**Comment by ChristophWurst** [comment]
My impression is that we don't have a last modification date for every contact.

If we don't have the data we should hide the text

**Comment by ChristophWurst** [comment]
https://github.com/nextcloud/server/issues/38544 for a related feature request. The SAB contacts do not have a REV at the moment.

**Comment by JohannesGGE** [comment]
I found where the magic happens:

https://github.com/nextcloud/contacts/blob/6f79fbc0f324925c8f1f5a048e4f98eb2ccf3898/src/services/checks/invalidREV.js#L55-L69

The fix is called because the fallowing returns true in some cases: 

https://github.com/nextcloud/contacts/blob/6f79fbc0f324925c8f1f5a048e4f98eb2ccf3898/src/services/checks/invalidREV.js#L31-L53

One of these cases are the vCard created by the contacts app:

```
BEGIN:VCARD
VERSION:4.0
PRODID:-//Nextcloud Contacts v5.3.0-beta2
UID:6aa9cf0f-e92e-4e5a-9c13-799ca265f9fd
FN:Contact
ADR;TYPE=HOME:;;;;;;
EMAIL;TYPE=HOME:
TEL;TYPE=HOME,VOICE:
REV;VALUE=DATE-AND-OR-TIME:20230531T135555Z
END:VCARD
```

`REV;VALUE=DATE-AND-OR-TIME:20230531T135555Z` seems to be a valid format for v4.0 of vCards but invalid for v3.0 as far as I understand it and as written in this issue https://github.com/nextcloud/contacts/issues/1352 . I don't understand why `version === '3.0' && type === 'vcardtime'` return true (valid but actually invalid) here, and 4.0 return false (invalid but it is actually valid) 

Ical.js says `revValue.icalclass === 'vcardtime'` for `REV;VALUE=DATE-AND-OR-TIME:20230531T135555Z`
and `revValue.icalclass === 'icaltime'` for `REV:20230531T135555Z`. Can't relate why.

---

So the actual problem is that the contacts app now writes vCard v4.0 and the fix wants to repair the valid REV. 
To fix the issue, I would just add the case `version === '4.0' && type === 'vcardtime'` where the validation returns valid. But as stated I don't really get the implementation here.

**Comment by ChristophWurst** [comment]
Does it also try to repair read-only system accounts?

I have to dig into details of vcard v3 and v4 too. @st3iny might know a few things too.

**Comment by JohannesGGE** [comment]
No, because SAB doesn't have REV yet, it will be set to now here:
https://github.com/nextcloud/contacts/blob/bf01de83fd643fc16b80d92f220d49ed7915dec0/src/models/contact.js#L84-L89

**Comment by st3iny** [comment]
Is hiding the rev if missing a valid workaround for now?

**Comment by ChristophWurst** [comment]
https://github.com/nextcloud/contacts/blob/830f17933904384a1ec5e78e9a3b2d5fba463cd5/src/components/ContactDetails.vue#L271 I think that code tries that but if the repair kicks in and sets a new REV it will render again

**Comment by realDayaa** [comment]
I can confirm that contacts with REV;VALUE=DATE-TIME show up correctly.

![REV;VALUE=DATE-TIME](https://github.com/user-attachments/assets/4fb9cdb7-386c-404a-8a98-3892468ca18f)

and contacts with REV;VALUE=DATE-AND-OR-TIME don't.

![REV;VALUE=DATE-AND-OR-TIME](https://github.com/user-attachments/assets/18aebb2d-d591-40c1-8ae8-4c7213b3cbae)

**Comment by st3iny** [comment]
Hi all, this should have been fixed by https://github.com/nextcloud/contacts/pull/4634 in the meantime. Please update the Contacts app and try again.
